### PR TITLE
Add CSV loader utility methods

### PR DESCRIPTION
**Pull request by bastien-boussouf**
I've been tinkering a bit with C# and bricks recently, just to test a few ideas. Since I wanted to use CSV files provided by Rebrickable, I came up with this proposal.

- Add new CSV loader classes
- Add themes method
- Move CHANGELOG to its ownfile
- Add launchSettings from demo project

I may have overstep a bit on the actual CSV features I wanted to add in this merge request. Feel free to tell me if that's too much and if I should reduce the scope or update a few things. For Instance, I haven't provided a factory pattern yet.

### CODE REVIEW

**Comment by gebirgslok** [comment]
Heyho Bastien,

first of all thank you very much for your contribution. Code looks very good and it adds good value to this lib. However, from a library maintainer perspective I see one pain point that we should get rid of.

With this new version every consumer of this lib would pull in two new dependencies (CsvHelper, System.Linq.Async) whether or not CSV utils are used (there is nothing wrong with them, but dependencies should be kept to a minimum).

I can think of two ways to solve this issue (open to further suggestions)

1. Don't use CsvHelper and use your own lightweight CSV parser
2. Make an additional NuGet package (e.g. RebrickableSharp.Csv) and let the user opt-in for CSV functionality (e.g. as it is done by EF Core with EntityFrameworkCore.SqlServer, ...Sqlite, etc.)

What do you think?

BR
Jens

**Comment by bastien-boussouf** [comment]
Hello Jens,

You raise a totally fair point. I would probably pick option 2 and publish an additional package as you suggested.

I'm not familiar with the Azure pipeline, but I'll give it a try. I'll update the pull request after moving the CSV related files to their own project.

**Comment by gebirgslok** [comment]
I can update the az pipeline once your PR got merged.

You need to find a way to get rod of the CsvColumn attributes from the Rebrickable entities.

Jens

**Comment by bastien-boussouf** [comment]
I've updated the branch with a new project and a different way on bringing json/csv compatibilty, it's a bit different now (54e740f).

I also pushed an attempt at some Azure updated pipeline with 29cdcfc. Given the fork nature, I'm not sure I can run it on a personal AzureDevOps project though.

**Comment by bastien-boussouf** [comment]
> I also pushed an attempt at some Azure updated pipeline with [29cdcfc](https://github.com/gebirgslok/RebrickableSharp/commit/29cdcfcadba6d7f8c667281927302e1498db579e). Given the fork nature, I'm not sure I can run it on a personal AzureDevOps project though.

Not a succesful attempt :) I'll undo this part, I guess you'll figure a better way.

File: RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs
Comment (gebirgslok) [inline RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs:14]: Should we accept an (optional) CancelationToken for all async methods in this class?

File: RebrickableSharp.Client.Csv/JsonCsvClassMap.cs
Comment (gebirgslok) [inline RebrickableSharp.Client.Csv/JsonCsvClassMap.cs]: I guess this class can be made internal?

File: RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs
Comment (bastien-boussouf) [reply RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs:14]: I guess we could yes, I'll take a look.

On a similar note, not sure that we would need to add more as the MeasureRequestAsync of the regular client. The http calls from the CsvLoader are fairly straight forward. What do you think?

File: RebrickableSharp.Client.Csv/JsonCsvClassMap.cs
Comment (bastien-boussouf) [reply RebrickableSharp.Client.Csv/JsonCsvClassMap.cs]: Probably yes. I'll do another pass at different modifiers.

File: RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs
Comment (bastien-boussouf) [reply RebrickableSharp.Client.Csv/RebrickableCsvLoader.cs:14]: 096c548e49a8e4dde9758a54b3add9d3b4277160

File: RebrickableSharp.Client.Csv/JsonCsvClassMap.cs
Comment (bastien-boussouf) [reply RebrickableSharp.Client.Csv/JsonCsvClassMap.cs]: c70e5eb6030008856b27a01a6ee63290cfea2d73

**Review by gebirgslok — APPROVED** [review]

**Comment by gebirgslok** [comment]
I'll Update the az pipeline in the next days. Thanks again for your contribution!

**Comment by gebirgslok** [comment]
> I'll Update the az pipeline in the next days. Thanks again for your contribution!

I just published v0.5.0 for both client and client.csv. :)

**Pull request merged**
